app/knight.py

```diff
@@ -0,0 +1,41 @@
+from typing import List, Dict, Optional
+
+
+class Knight:
+    def __init__(
+            self,
+            name: str,
+            power: int,
+            hp: int,
+            armour: List[Dict[str, int]],
+            weapon: Dict[str, int],
+            potion: Optional[Dict[str, Dict[str, int]]] = None
+    ) -> None:
+        self.name = name
+        self.base_power = power
+        self.base_hp = hp
+        self.armour = armour
+        self.weapon = weapon
+        self.potion = potion
+        self.apply_armour()
+        self.apply_weapon()
+        self.apply_potion()

```

Comment:
Let's move these setup functions to be running from another function like `setup()` and then this function will run these 3